### ISSUE TITLE
[Bug]: Bitbucket Previews are not updated on push

### ISSUE THREAD

**Issue by lorenzomigliorero** [coollabsio/coolify#2793]
### Description

Pushing to a PR branch doesn't update the Preview deployment.
This is the response from _/webhooks/source/bitbucket/events/manual_:

```json
{"status":"failed","message":"Nothing to do. No applications found with deploy key set, branch is '***' and Git Repository name has ***"}
```

### Minimal Reproduction

- Create an application with BitBucket repository as its source
- Set up the BitBucket webhook && enable preview deployments
- Create PR in the repo
- Push to the PR branch
- Preview deployment isn't updated

### Version

v4.0.0-beta.307

**Comment by Xiloe** [comment]
Yes, coolify doesn't support the pullrequest:updated request from bitbucket, I'll see if I can do a PR.
![image](https://github.com/coollabsio/coolify/assets/51050533/4ebf7215-341a-4810-a2df-29f3ff213a6f)

**Comment by lorenzomigliorero** [comment]
@Xiloe Thanks for the answer! Actually, a manual PR redeploy always fetches the HEAD instead of the PR branch. I opened another issue here https://github.com/coollabsio/coolify/issues/2792.

**Comment by Xiloe** [comment]
I'll look at this later tomorrow, seems like it's the same issue that was going on with gitea that I fixed and got merged in version 308

**Comment by IARayan** [comment]
Submitted a fix and I would like to explain what exactly was happening here since this is the related bug and I spent a good couple of days debugging this issue (Since, we are using bitbucket as well)

This happens because normally when a deployment is created by either a manual deployment or if you have the preview deployments enabled, by taking the branch name and git repository name and saving them in the `applications` table, that's how Coolify stores information about it's applications and if it's a preview deployment, then it adds an entry to the `application_previews` table

Now the interesting part is, BitBucket's repo:push webhook payload, doesn't contain information that determines if this commit is a part of a Pull Request, and hence it doesn't also include the destination branch name

Currently Coolify tries to get the branch name of repo:push by accessing the payload and getting the value of this element `push.changes.0.new.name` this returns the branch that this commit was added to but nothing related to destination
by switching to pullrequest:update we are able to access the destination branch name (which matches what we have in the `applications` table) and correctly find that application and update it's preview deployment if it exists.

Now as I'm writing this I realize that we also might use repo:push to update the main branch if we have an application that matches it, which instructs me to update the PR again, for backward compatibility.

**Comment by IARayan** [comment]
The repo:push payload lacks the destination branch name, using push.changes.0.new.name instead. While this works for updates, it causes failures in preview deployments when a PR receives new commits. Coolify looks for git_branch in the applications table, but instead, it finds the source branch name, causing issues in preview deployment updates.

By also processing pullrequest:updated, we ensure that preview deployments remain in sync even when PRs are updated. However, this event triggers on multiple actions (e.g., PR title changes, reviewer updates), which results in extra redeploys. Since Coolify doesn’t store commit hashes for preview deployments, handling these extra redeploys is the best workaround for Bitbucket.
